manager/integration/tests/test_ha.py

```diff
@@ -2620,107 +2620,145 @@ def test_replica_failure_during_attaching(settings_reset, client, core_api, volu
        node-1 since there is no engine image on node-1. The attach API call
        returns error
-    8. Verify that we can attach to another node, take snapshot, take a backup,
+    9. Verify that we can attach to another node, take snapshot, take a backup,
        expand, then detach vol-1
-    9. Verify that vol-2 cannot be attached to any nodes because one of
+    10. Verify that vol-2 cannot be attached to any nodes because one of
        its replicas is sitting on the node-1 which doesn't have the
        engine image. The attach API call returns error
+    """
+    pass
+
 
-    Case 3: Test engine upgrade when engine image DaemonSet is not fully
+@pytest.mark.skip(reason="TODO") # NOQA
+def test_engine_image_not_fully_deployed_perform_engine_upgrade():
+    """
+    Test engine upgrade when engine image DaemonSet is not fully
     deployed
-    1. Continue from case #2
-    2. Deploy a new engine image, new-ei
-    3. Detach vol-1
-    4. Verify that you can upgrade vol-1 to new-ei
-    5. Attach vol-1 to node-2
-    6. Verify that you can live upgrade vol-1 to back to default engine image
-    7. Try to upgrade vol-2 to new-ei
-    8. Verify that the engineUpgrade API call returns error
-
-    Case 4: Test replicas scheduling when engine image DaemonSet is not fully
+
+    1. Create volume 1 and volume 2 with 3 replicas
+    2. Taint node-1 with the taint: key=value:NoSchedule
+    3. Attach vol-1 to node-1. Change the number of replicas of vol-1
+       to 2. Delete the replica on node-1
+    4. Delete the pod on node-1 of the engine image DaemonSet.
+       Or delete the engine image DaemonSet and wait for Longhorn
+       to automatically recreates it.
+    5. Wait for the engine image CR state become deploying
+    6. Deploy a new engine image, new-ei
+    7. Verify that you can upgrade vol-1 to new-ei
+    8. Detach then attach vol-1 to node-2
+    9. Verify that you can live upgrade vol-1 to back to default engine image
+    10. Try to upgrade vol-2 to new-ei
+    11. Verify that the engineUpgrade API call returns error
+    """
+    pass
+
+
+@pytest.mark.skip(reason="TODO") # NOQA
+def test_engine_image_not_fully_deployed_perform_replica_scheduling():
+    """
+    Test replicas scheduling when engine image DaemonSet is not fully
     deployed
-    1. Continue from case #3
-    2. Create a new volume, vol-3, with 2 replicas
-    3. disable the scheduling for node-2
-    4. Verify that there is one replica fail to be scheduled
-    5. enable the scheduling for node-2
-    6. Verify that replicas are scheduled onto node-2 and node-3
-
-    Case 5: Test auto upgrade engine feature when engine image DaemonSet is
+
+    1. Taint node-1 with the taint: key=value:NoSchedule
+    2. Delete the pod on node-1 of the engine image DaemonSet.
+       Or delete the engine image DaemonSet and wait for Longhorn
+       to automatically recreates it.
+    3. Wait for the engine image CR state become deploying
```

Comment:
Can we make these steps as prerequisite for the test and rest below tests?

Reply:
Updated!